src/idlcxx/src/backendCpp11Trait.c

```diff
@@ -87,14 +87,32 @@ generate_traits(idl_backend_ctx ctx, const idl_node_t *node)
 static idl_retcode_t
 generate_macro_call(idl_backend_ctx ctx, const idl_node_t *node)
 {
-  assert(node->mask & IDL_STRUCT);
-
-  char *struct_name = get_cpp11_fully_scoped_name(node);
+  if (node->mask & IDL_STRUCT)
```

Comment:
Should this be (node->mask & IDL_STRUCT) == IDL_STRUCT, since this may cause issues if IDL_STRUCT consists of more than 1 byte set.
There is also a utility function in libidl called `bool idl_is_struct(idl_node_t *node)` testing exactly this thing.

Reply:
@reicheratwork Given the fact that each type only represents one bit, the additional check would technically not be needed. And given the fact that the system was designed to have one bit per category, I don't think this will change any time soon. However, if we do think this should be avoided, we should do so not only in the backendCpp11Trait.c file, but also in the backendCpp11Type.c file.

Reply:
That would be preferred, so I agree with @reicheratwork here, but I'm happy to accept it for now. Probably best to use the functions in the future though, they do some extra checks, so it's easier to spot errors/bugs.